conans/client/generators/msbuild.py

```diff
@@ -87,6 +87,10 @@ def _deps_props(self, name_general, deps):
         template = textwrap.dedent("""\
             <?xml version="1.0" encoding="utf-8"?>
             <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+                <PropertyGroup Label="Conan-PackageInfo">
```

Comment:
I am reviewing this, and while it seems correct, I might be missing something.

All other variables are appended with the package name like ``includedirs_pkgname``, because otherwise different packages will overwrite each other values.

The generated .props files will include transitively other .props files if there are transitive dependencies, which is the common case. In this case the ``ConanPackageName`` and ``ConanPackageVersion`` properties will be overwritten, and only 1 value accesible.

It would be necessary to have a complete case, in which the consumption, usage pattern is shown. Why and how a consumer will need this data? Wouldn't it need the data from all transitive dependencies that are being included?

I would say yes, in that case, the solution is a bit more complicated, and we would need at least some cumulative variable or something similar. 

Please tell me what you think.

Reply:
James, thank you for your input. I think you raise a valid concern.

My goal is to recover something analogous to:
https://github.com/conan-io/conan/blob/4e3df5fc30bdb63936ab8f196ab8fb56068c511a/conans/client/generators/cmake.py#L94-L95

From the `CMake` generator. 

I could be mistaken, but I don't think the `CMake` generator worries about the transitive dependencies either.

My goal was to generate `ConanPackageName` and `ConanPackageVersion` only once in the top level property file.

Reply:
Our real use case, is not unsurprisingly, a ~~little~~ lot more complicated.

We have a single top level `conanfile.py` that captures the shared build/runtime requirements for a variety of tools/utilities used for the development of embedded software -- think something akin to [build-essential](https://packages.ubuntu.com/xenial/build-essential) but with a lot of our home grown utilities.

To produce these tools, we have a single top level Visual Studio solution with a variety of sub projects that depend on these build / runtime requirements. Each sub project imports the generated `conanbuildinfo.props`.

One project is for a tool that produces artifacts for troubleshooting / debugging embedded  software. Another project contains a mock/example embedded code that we compile and then produce build artifacts for.

We would like to package the artifacts for our example project with the same version as the high level project that produced them -- that way we can say `PkgBuildEssential@August2020` produced `PkgExampleCode@August2020` and keep the two in sync.

This lets downstream tools consuming these example build artifacts and obtain good reference files from the latest "internal build essentials" release.

A good spot for pinning a top level version seemed to be the top level `conanfile.py` for the project. And we could get there if the `visual_studio` and `msbuild` generators behaved more like the `cmake` generator and adding something analogous to `CONAN_PACKAGE_NAME` and `CONAN_PACKAGE_VERSION` into one of the generated property files.

Reply:
According to the doc:
> One conan_deps.props Visual Studio properties file, importing all the direct dependencies, in this example both conan_zlib.props and conan_poco.props.

`conan_deps.props` seemed like the top level property file. But a different property file would work just as well if it's a bad fit logically / conceptually.

_In fact, that's how we're currently working around the issue -- by defining a custom generator that produces yet another property file containing this top level information._

Reply:
> Oh, I think I understand. It seems that you want the name + version of the current package, not the dependencies, and this is why I was probably confused. Did I understand it right now?

Correct. It sounds like we're on the same page now. I hadn't looked too closely at the new `msbuild` generator and didn't see any references to the toolchain capabilities from the generator page. This must've been what was being [hinted at](https://github.com/conan-io/conan/issues/7177#issuecomment-643137653) earlier.  Sorry I didn't make the connection.

> In that case, the place to implement this is most likely the toolchain(), and the MSBuildToolchain(). It is experimental, but it totally sounds like the right place to be defined. Have you had a chance to have a look at them?

conan_toolchain.props does seem like a better fit. Would you rather I put ` <PropertyGroup Label="Conan-PackageInfo">` there?

Reply:
Hi @thorntonryan

yes, definitely, I suggest moving this name-version info to the generated ``conan_toolchain.props`` from the ``MSBuildToolchain()``, this is the new proposed integration with build systems and where we are putting our efforts.